_opensearch/search-backpressure.md

```diff
@@ -0,0 +1,217 @@
+
+## Search backpressure settings
+
+Search backpressure adds several settings to the standard OpenSearch cluster settings. These settings are dynamic, so you can change the default behavior of this feature without restarting your cluster.
+
+Setting | Default | Description
+:--- | :--- | :---
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;mode | `monitor_only` | The [mode](#search-backpressure-modes) for search backpressure. Valid values are `monitor_only`, `enforced`, or `disabled`.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;interval | 1 second | The interval at which the observer thread measures the resource consumption and cancels tasks.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;cancellation_ratio | 10% | The maximum percentage of tasks to cancel out of the number of successful task completions.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;cancellation_rate | 0.003 | The maximum number of tasks to cancel per millisecond.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;cancellation_burst | 10 | The maximum number of tasks that can be canceled before no further cancellations are made.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;node_duress.<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;num_consecutive_breaches | 3 | The number of consecutive limit breaches after which the node is marked in duress.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;node_duress.<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cpu_threshold | 90% | The CPU usage threshold (in percentage) for a node to be considered in duress.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;node_duress.<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;heap_threshold | 70% | The heap usage threshold (in percentage) for a node to be considered in duress.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;search_heap_threshold | 5% | The heap usage threshold (in percentage) for the sum of heap usages across all search tasks before server-side cancellation is applied.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;search_task_heap_threshold | 0.5% | The heap usage threshold (in percentage) for one task before it is considered for cancellation.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;search_task_heap_variance | 2 | The heap usage variance for one task before it is considered for cancellation. A task is considered for cancellation when `taskHeapUsage` is greater than or equal to `heapUsageMovingAverage` &middot; `variance`.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;search_task_cpu_time_threshold | 15 seconds | The CPU usage threshold (in milliseconds) for one task before it is considered for cancellation.
+search_backpressure.<br>&nbsp;&nbsp;&nbsp;&nbsp;search_task_elapsed_time_threshold | 30 seconds | The elapsed time threshold (in milliseconds) for one task before it is considered for cancellation.
```

Comment:
Some of these settings have been renamed. Sharing the latest ones along with simplified descriptions.

* `search_backpressure.mode` (default monitor_only)
   The search backpressure [mode](#search-backpressure-modes). Valid values are `monitor_only`, `enforced`, or `disabled`.
   
* `search_backpressure.interval_millis` (default 1 second)
  The interval at which the observer thread measures the resource usage and cancels tasks.
  
* `search_backpressure.cancellation_ratio` (default 10%)
   The maximum number of tasks to cancel as a fraction of successful task completions.

* `search_backpressure.cancellation_rate` (default 0.003)
   The maximum number of tasks to cancel per millisecond of elapsed time.

* `search_backpressure.cancellation_burst` (default 10)
   The maximum number of tasks to cancel in a single iteration of the observer thread.

* `search_backpressure.node_duress.num_successive_breaches` (default 3)
   The number of of successive limit breaches after which the node is considered under duress.

* `search_backpressure.node_duress.cpu_threshold` (default 90%)
   The CPU usage threshold (in percentage) for a node to be considered in duress.

* `search_backpressure.node_duress.heap_threshold` (default 70%)
   The heap usage threshold (in percentage) for a node to be considered in duress.

* `search_backpressure.search_shard_task.total_heap_percent_threshold` (default 5%)
   The heap usage threshold (in percentage) for the sum of all search shard tasks before cancellation is applied.

* `search_backpressure.search_shard_task.heap_percent_threshold` (default 0.5%)
   The heap usage threshold (in percentage) for a single search shard task before it is considered for cancellation.

* `search_backpressure.search_shard_task.heap_variance` (default 2.0)
   The minimum variance of a single search shard task's heap usage usage compared to the rolling average of previously completed tasks before it is considered for cancellation.

* `search_backpressure.search_shard_task.heap_moving_average_window_size` (default 100)
   The number of previously completed search shard tasks to consider when calculating the moving average of heap usage.

* `search_backpressure.search_shard_task.cpu_time_millis_threshold` (default 15 seconds)
   The CPU usage threshold (in milliseconds) for a single search shard task before it is considered for cancellation.

* `search_backpressure.search_shard_task.elapsed_time_millis_threshold` (default 30 seconds)
   The elapsed time threshold (in milliseconds) for a single search shard task before it is considered for cancellation.